Documentation/WritingReST/Codeblocks.rst

```diff
@@ -15,185 +15,265 @@ Code blocks with syntax highlighting
 
+            /**
+             * Add default TypoScript (constants and setup)
+             */
+            \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addStaticFile(
+                'site_package',
+                'Configuration/TypoScript',
+                'Site Package'
+            );
 
-      How this looks:
-         See following example:
+   .. group-tab:: Output
 
-         .. code-block:: php
+      .. code-block:: php
+         :caption: EXT:site_package/Configuration/TCA/Overrides/sys_template.php
+
+         /**
+          * Add default TypoScript (constants and setup)
+          */
+         \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addStaticFile(
+             'site_package',
+             'Configuration/TypoScript',
+             'Site Package'
+         );
 
-            $a = 'b';
+Always use :ref:`syntactically correct code <codeblocks-syntactically-correct>`
+in a code block.
 
-      You can use this, if the default language is already set to PHP with the :ref:`highlight
-      directive <codeblocks-highlight-directive>` in the current file
-      (or in :ref:`Includes.rst.txt <includes-rst-txt>`).
+Use :ref:`placeholders <codeblocks-placeholders>` in angle brackets
+(`<placeholder-name>`) to refer to a place in the code where you don't care
+about the exact value.
 
 
+.. index:: reST directives; code-block
+.. _codeblock-directive:
 
-   2. Set the language (PHP) in the `code-block`::
+Code block directive
+====================
 
-         See following example:
+A code block has the following syntax:
 
-         .. code-block:: php
+.. code-block:: rst
+   :caption: Documentation/MyDocs.rst
 
-            $a = 'b';
+   .. code-block:: <language>
+      :caption: <caption>
+      <options>
 
+      <code>
 
-   3. Use `code-block` without setting the language::
+There must be a new line between the options and the code. There must not be a
+new line between options or options and directive line. Otherwise the rendering
+will fail.
 
-         See following example:
+<language>
+   The language of the code-block. We commonly use the following: :rst:`php`,
+   :rst:`typoscript` (also for tsconfig), :rst:`yaml`, :rst:`javascript`,
+   :rst:`html` (also for Fluid), :rst:`shell`, :rst:`xml`, :rst:`none`.
 
-         .. code-block::
+   See all
+   :ref:`available lexers <writing-rest-codeblocks-available-lexers>`
 
-            $a = 'b';
+<caption>
+   To make orientation in code-examples easier we try to always add
+   a caption with the name of the file where the example should go. If the code
+   should be entered in the backend explain where it should go there.
 
-      You can use this, if you already set the language PHP with the :ref:`highlight
-      directive <codeblocks-highlight-directive>` in the current file
-      (or in :ref:`Includes.rst.txt <includes-rst-txt>`).
+<options>
+   The following options may be used, all options but :rst:`caption` are
```

Comment:
Caption is optional, too. Am I wrong?

Reply:
By Syntax yes, But we want to always use it.

Reply:
OK, but then we should also phrase it like this. This is already done in the other sections but here it sounds like the there is a technical reason why the caption is required.

Reply:
How would you rephrase it?

Reply:
See suggestion.